tests/zfs-tests/tests/functional/events/events_common.kshlib

```diff
@@ -41,41 +66,35 @@ function file_wait # file delay timeout
 
 	while [ $(( $(date +%s) - $(stat -c %Y $file) )) -lt $delay ]; do
 		if [[ $SECONDS -gt $timeout ]]; then
+			echo file_wait exceeded $SECONDS seconds
 			return 1
 		fi
 
 		sleep 1
+		((SECONDS=SECONDS+1))
 	done
 
 	return 0;
 }
 
 function run_and_verify
 {
-	typeset delay event pool zedlog
+	typeset event pool
 	set -A events
 
 	while getopts "d:e:p:z:" opt; do
 		case $opt in
-		d)
-			delay=$OPTARG
-			;;
```

Comment:
Does this need to get added back due to the use of `-d` below?

Reply:
ah, good catch.  '-d' isn't used in that context (it will use `file_wait_event` now).  That addition was from an earlier revision and can be rolled back out.